chainerx_cc/chainerx/python/array.cc

```diff
@@ -85,6 +86,29 @@ py::array MakeNumpyArrayFromArray(const ArrayBodyPtr& self, bool copy) {
     return py::array{dtype, shape, strides, ptr, py::cast(internal::MoveArrayBody(std::move(array)))};
 }
 
+py::object MakeCupyArrayFromArray(const py::module& m, py::handle self) {
+    Array array = Array{py::cast<ArrayBodyPtr>(self)};
+    const Device& device = array.device();
+    // TODO(okapies): rejects if array's device is not compatible with cupy
```

Comment:
I have no idea how to check whether the device is compatible with CuPy or not...